Array translations

joedixon opened the pull request:
This PR adds support for the array style languages files which should resolve #17 

In order to facilitate this, I've added a new dropdown item to the nav which lists all of the files for the selected language. Selecting any filename will filter the keys available for translation.

If the value of a line in the array language file is an array, a textarea is rendered for each of the keys in that array. This will only work one level deep.

![screen shot 2017-10-09 at 14 08 56](https://user-images.githubusercontent.com/3438564/31339661-7dc96602-acfb-11e7-88fa-585b1724ecad.png)

As part of this update, a default language file is created if none already exists with sample translation. This should fix #27

joedixon committed:
Order imports by size

joedixon committed:
Refactor collection of JSON translations to method

joedixon committed:
Get array translations

joedixon committed:
Refactor to support array translations

joedixon committed:
Refactor for array translations

joedixon committed:
Formatting

joedixon committed:
Add translation files to menu

joedixon committed:
Display multiple textarea inputs if value is an object

joedixon committed:
Refactor to watch methods

Doing this fixes the issue where the save button state did not reset
after clicking save. It also prevents the reload dialog if changes have
been saved

joedixon committed:
Refactor to suport array translations

joedixon committed:
Rebuild public js

joedixon committed:
Create default language if it doesn't exist

joedixon committed:
Create default language if not exists

joedixon committed:
Merge branch 'master' into array-translations

joedixon committed:
Revert removal of environment variable

ubisend commented:
I have an update coming for this which will update the scanning method to include file.key syntax used in array files so probably not worth merging until it's done.

joedixon commented:
I have an update coming for this which will update the scanning method to include file.key syntax used in array files so probably not worth merging until it's done.

joedixon committed:
Use watchers to determine when translations array is updated

joedixon committed:
Rebuild public js

joedixon committed:
Refactor sync method

Look for strings and groups and sync both accordingly

Return the full translations array

joedixon committed:
Extract string and group sync methods

joedixon committed:
Convert null to empty string

Necessitated by the ConvertEmptyStringsToNull middleware.

Prevented texture from rendering on the frontend

joedixon committed:
Capture strings and groups

joedixon committed:
Extract methods

joedixon committed:
Refactored

joedixon committed:
Convert null to empty strings

joedixon committed:
Update formatting

joedixon committed:
Add watcher for selectedKey

Without this, selecting the 0 index of a numerically indexed
translation file will not render the texture

joedixon committed:
Rebuild JS

joedixon commented:
I have updated this PR so that scanning now includes array style translation files

themsaid commented:
Thank you, will be looking into these two with the next release in the upcoming few weeks.

joedixon commented:
No problem, happy to chat through it if needed :-)

jmarcher commented:
Is there a way to port this to the actual version?
